src/libraries/System.Net.WebSockets.Client/tests/DeflateTests.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Test.Common;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.WebSockets.Client.Tests
+{
+    public class DeflateTests : ClientWebSocketTestBase
+    {
+        public DeflateTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [ConditionalTheory(nameof(WebSocketsSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34690", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/42852", TestPlatforms.Browser)]
```

Comment:
These aren't supported on the browser so probably shouldn't need an issue associated with them.

Reply:
I am unsure about this one. WebSocketsSupported is defined as:
```c#
if (PlatformDetection.IsBrowser && !PlatformDetection.IsBrowserDomSupported)
{
     return false;
}
```
Which leads to to believe that there are some WASM environments where it should be supported. I've copied this from other tests in the same project that were using LoopbackServer. See https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs#L375-L379 for example.

Let me know if I'm wrong and I will remove the linked issue for the Browser case.

Reply:
In this case, there is no support in the browser for client options other than sub protocol, so these options can never work (unless browsers add new features)
https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

cc @lewing in case I'm wrong about the usage of `ActiveIssue` here.

Reply:
What about for example, Blazor Desktop? Having a wasm environment without a browser, just a JavaScript engine. Would it make sense in this case that we should be able to use ClientWebSocket with its full capabilities, or I am completely wrong :)?

Reply:
> Blazor Desktop? Having a wasm environment without a browser, just a JavaScript engine

I haven't been following that work at all, but I would hope `TestPlatforms.Browser` doesn't apply in that case :)

Reply:
You can remove the ActiveIssue for the Browser case and just skip the tests on Browser. Desktop blazor uses the desktop runtime.